catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -517,6 +520,164 @@ public void init() {
             return "";
         });
 
+        /*
+         * Notes may have a virtual parent, which means the parent is not directly accessible to
+         * the framework. For example, the parent may be stored as a text field inside another
+         * metacard. For non-virtual parents, the parent must be retrievable by the framework.
+         * If the call does not set the isVirtual flag, then it defaults to 'false'.
+         */
+        post("/notes", (req, res) -> {
+            Map<String, Object> incoming = JsonFactory.create()
+                    .parser()
+                    .parseMap(req.body());
+            String parentID = incoming.get("parent")
+                    .toString();
+            String comment = incoming.get("note")
+                    .toString();
+            Boolean isVirtual = Boolean.parseBoolean(incoming.getOrDefault("isVirtual", "false")
+                    .toString());
+            String user = getSubjectEmail();
+            if (user == null) {
+                return util.getResponseWrapper("error",
+                        "You are not authorised to create notes! A user email is required. "
+                                + "Please ensure you are logged in and/or have a valid email registered in the system.");
+            }
+            //Test if parent metacard ID is valid and comment is not empty
+            if (!isVirtual) {
+                try {
+                    util.getMetacard(parentID, true);
+                } catch (NotFoundException e) {
+                    LOGGER.debug("Parent metacard ID is invalid!");
+                    throw new IllegalStateException("Parent metacard ID is invalid!");
+                }
+            }
+            if (comment.equals("")) {
+                LOGGER.info(util.getResponseWrapper("error", "No comment!"));
+                return util.getResponseWrapper("error", "No comment!");
+            }
+            MetacardType noteType = null;
+            for (MetacardType type : types) {
+                if (type.getName()
+                        .equals("resource-note")) {
+                    noteType = type;
```

Comment:
I tried make it a service, but apparently you can't create a service and reference that service in the same pom. But I did create a service so other can use the metacard type, and I just pass the actual metacard type into the ctor.